### PR TITLE
Cypress UI Test service and updates to installer.py

### PR DESCRIPTION
**Pull request by varun-krishnamurthy**
> Please provide enough information so that others can review your pull request:

Provides solution for feature requests #1247 and #1248 1248

> Explain the **details** for making this change. What existing problem does the pull request solve?

1. Changes to docker-compose to use full qualified repo url (docker.io).
2. Updates to installer.py to include --db-type argument and related functions
3. Updates to relevant documentation.
4. New utility script install_x11_deps.sh to install X11 deps on Linux and Mac.

### CODE REVIEW

File: development/installer.py [hunk 1]
```diff
@@ -129,6 +136,7 @@ def init_bench_if_not_exist(args):
         init_command += f"--frappe-branch={args.frappe_branch} "
         init_command += f"--apps_path={args.apps_json} "
         init_command += args.bench_name
+        print(init_command)
```
Comment (revant) [inline development/installer.py:139]: Remove print

File: development/installer.py [hunk 2]
```diff
@@ -188,20 +199,42 @@ def init_bench_if_not_exist(args):
...
+        )
+        new_site_cmd = [
+            "bench",
+            "new-site",
+            f"--db-host=postgresql",  # Should match the compose service name
+            f"--db-type={args.db_type}",  # Add the selected database type
+            f"--db-root-password=123",  # Replace with your PostgreSQL password
+            f"--admin-password={args.admin_password}",
+        ]
+    print(new_site_cmd)
```
Comment (revant) [inline development/installer.py:232]: Remove print statement

File: docs/development.md [hunk 3]
```diff
@@ -241,15 +241,14 @@ Sample `apps-example.json` is used by default, it installs erpnext on current st
...
+python installer.py  #pass --db-type postgres for postgresdb
 ```
 
 For command help
 
 ```shell
 python installer.py --help
-usage: installer.py [-h] [-j APPS_JSON] [-b BENCH_NAME] [-s SITE_NAME] [-r FRAPPE_REPO] [-t FRAPPE_BRANCH] [-p PY_VERSION] [-n NODE_VERSION] [-v]
-
+usage: installer.py [-h] [-j APPS_JSON] [-b BENCH_NAME] [-s SITE_NAME] [-r FRAPPE_REPO] [-t FRAPPE_BRANCH] [-p PY_VERSION] [-n NODE_VERSION] [-v] [-a ADMIN_PASSWORD] [--db-type DB_TYPE]
```
Comment (revant) [inline docs/development.md:251]: Re print usage with new options -d --db-type

File: docs/development.md [hunk 4]
```diff
@@ -380,3 +382,25 @@ volumes:
...
 Access the service by service name from the `frappe` development container. The above service will be accessible via hostname `postgresql`. If ports are published on to host, access it via `localhost:5432`.
+
+## Using Cypress UI tests
+
+To run cypress based UI tests in a docker environment, follow the below steps:
+
+1. Install and setup X11 tooling on VM using the script `install_x11_deps.sh`
+
+```shell
+  sudo ./install_x11_deps.sh
```
Comment (revant) [inline docs/development.md:393]: sudo bash install_x11_deps.sh

It'll ensure execution even if file doesn't have exec permission

**Review by revant — CHANGES_REQUESTED** [review]
Fix pre-commit lint

File: README.md [hunk 5]
```diff
@@ -46,7 +46,6 @@ Wait for 5 minutes for ERPNext site to be created or check `create-site` contain
 - [Development using containers](docs/development.md)
 - [Bench Console and VSCode Debugger](docs/bench-console-and-vscode-debugger.md)
 - [Connect to localhost services](docs/connect-to-localhost-services-from-containers-for-local-app-development.md)
-
```
[the remark this answers is not in the document]
Comment (varun-krishnamurthy) [reply README.md]: fixed

**Comment by varun-krishnamurthy** [comment]
> Fix pre-commit lint


> Fix pre-commit lint

**Pull request merged**